Stratis.Bitcoin.Features.Consensus/StakeItem.cs

```diff
@@ -1,13 +1,84 @@
 ﻿using NBitcoin;
+using Stratis.Bitcoin.Utilities;
 
 namespace Stratis.Bitcoin.Features.Consensus
 {
-    public class StakeItem
+    /// <summary>
+    /// An object that holds block stake information.
+    /// </summary>
+    public sealed class StakeItem
     {
-        public uint256 BlockId;
-        public BlockStake BlockStake;
-        public bool InStore;
-        public long Height;
-    }
-}
+        private StakeItem() { }
+
+        /// <summary>The hash of the block.</summary>
+        public uint256 BlockHash { get; private set; }
+
+        /// <summary>The block stake.</summary>
+        public BlockStake BlockStake { get; private set; }
+
+        /// <summary>The height of the block.</summary>
+        public long? Height { get; private set; }
+
+        /// <summary>
+        /// Returns <c>true</c> if the item exists in the store. It is used to determine
```

Comment:
I wouldn't use `Returns` here as this is rather a property, yes its getter does return, but I'd speak about it as a value, so just `<c>true</c> if the item exists in the store. ...`